core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -526,6 +527,25 @@ private ExecutionModel parseCreateTable(GenericLexer lexer, SqlExecutionContext
             }
             model.setPartitionBy(partitionBy);
             tok = optTok(lexer);
+
+            if (tok != null) {
+                if (isWalKeyword(tok)) {
+                    if (!PartitionBy.isPartitioned(model.getPartitionBy())) {
+                        throw SqlException.position(lexer.getPosition()).put("WAL Write Mode can only be used on partitioned tables");
+                    }
+                    walSetting = walEnabled;
+                    tok = optTok(lexer);
+                } else if (isBypassKeyword(tok)) {
+                    tok = optTok(lexer);
+                    if (tok != null && isWalKeyword(tok)) {
+                        walSetting = walDisabled;
+                        tok = optTok(lexer);
+                    } else {
+                        throw SqlException.position(lexer.getPosition()).put(" invalid syntax, should be BYPASS WAL but was BYPASS " + tok);
```

Comment:
incorrect error position, this will position cursor on XYZ in `bypass something xyz`. The cursor should be pointing at incorrect keyword (`lastTokenPosition`). In this case `null` should also be treated differently.

Reply:
fixed in https://github.com/questdb/questdb/pull/2335/commits/bf37ae5577a61e21bd8604025c24c6259b0cca1a